src/omg_dosimetry/analysis.py

```diff
@@ -653,8 +672,34 @@ def plot_profile(self, ax=None, profile='x', position=None, title=None, diff=Fal
             diff_prof = film_prof - ref_prof
             ax_diff.set_ylabel("Difference (cGy)")
             ax_diff.plot(x_axis, diff_prof,'g-', linewidth=0.25)
+            
+        if vertical_line:
+            ax.plot((vertical_line / self.film_dose.dpmm, vertical_line / self.film_dose.dpmm), ax.get_ylim(), 'k:', linewidth = 1)
+            
+        if xlim:
```

Comment:
I needed those for the improved PDF reports. When "clusters analysis" is used, I generate new pages showing the analysis on individual clusters. I plot isodoses and profile plots that are focused on the cluster (instead of showing profiles that span the whole film).